File: Frontend/src/app/Core/service/model/dish.ts
export class dish{
  dishId : string;
  dishName : string;
  dishDescription : string;
  dishPriceValue :number;
  dishStatus : string;
  dishImageUrl : string;
  dishTypeName : string;
  restaurant : string;
  dishSalePrice : number;
  restaurantId : number
  constructor() {
    this.dishId = '';
    this.dishName = ''
    this.dishDescription = ''
    this.dishPriceValue = 0
    this.dishStatus = ''
    this.dishImageUrl = ''
    this.dishTypeName = ''
    this.restaurant = ''
    this.dishSalePrice = 0
    this.restaurantId  = 0
  }

}
export class dishType{
  dishTypeId : number | null = null;
  dishTypeName : string | null = null;
  dishTypeDescription : string | null = null;
  dishTypeCreatedAt : string | null = null;
  dishTypeCreatedBy : string | null = null;
  dishTypeUpdatedAt : string | null = null;
  dishTypeUpdatedBy : string | null = null;
}

File: Backend/foodee_service/src/main/java/org/example/foodee_service/entity/Dish.java
package org.example.foodee_service.entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.example.foodee_service.util.SecurityUtil;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Dish {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long dishId;

    @Column(nullable = false)
    private String dishName;

    @Column(columnDefinition = "TEXT")
    private String dishDescription;

    @Column(precision = 10, scale = 2)
    private BigDecimal dishPriceValue;

    @Column(precision = 10, scale = 2)
    private BigDecimal dishSalePrice;

    @Column(length = 50)
    private String dishStatus;

    private String dishImageUrl;

    private LocalDateTime dishCreatedAt;

    private String dishCreatedBy;

    private LocalDateTime dishUpdatedAt;

    private String dishUpdatedBy;

    @ManyToOne
    private DishType dishType;

    @ManyToOne
    private Restaurant restaurant;

    @PrePersist
    public void prePersist() {
        dishCreatedAt = LocalDateTime.now();
        dishCreatedBy = SecurityUtil.getCurrentUsername();
    }

    @PreUpdate
    public void preUpdate() {
        dishUpdatedAt = LocalDateTime.now();
        dishUpdatedBy = SecurityUtil.getCurrentUsername();
    }
}

